pkg/core/dao/dao.go

```diff
@@ -809,12 +840,86 @@ func (dao *Simple) getDataBuf() *io.BufBinWriter {
 // Persist flushes all the changes made into the (supposedly) persistent
 // underlying store. It doesn't block accesses to DAO from other threads.
 func (dao *Simple) Persist() (int, error) {
+	dao.persistNativeCache()
 	return dao.Store.Persist()
 }
 
 // PersistSync flushes all the changes made into the (supposedly) persistent
 // underlying store. It's a synchronous version of Persist that doesn't allow
 // other threads to work with DAO while flushing the Store.
 func (dao *Simple) PersistSync() (int, error) {
+	dao.persistNativeCache()
```

Comment:
We have now updated cache and stale storage between these lines.

Reply:
Thats a tricky one, because we can't use the approach from memcached store where the local store is created, because our cache always needs properly initialized backend cache. On the other hand, we can't hold cachelock within the whole PersistSync/Persist, because it will block other threads from accessing the whole DAO.

I had an idea to create a copy of backend cache to use it during main cache persist, but it's not cheap. I'll think of it.